src/main/kotlin/org/rust/lang/core/macros/MacroDiagram.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.macros
+
+import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.rust.lang.core.psi.RsMacro
+import org.rust.lang.core.psi.RsMacroBinding
+import org.rust.lang.core.psi.RsMacroBindingGroup
+import org.rust.lang.core.psi.ext.fragmentSpecifier
+
+enum class FragmentKind {
+    Ident,
```

Comment:
Enum items should be in CAPS as far as I know...

Reply:
> For enum constants, it's OK to use either uppercase underscore-separated names (enum class Color { RED, GREEN }) or regular camel-humps names starting with an uppercase letter, depending on the usage.

https://kotlinlang.org/docs/reference/coding-conventions.html